### DISCUSSION TITLE
How to programmatically disconnect a wallet/account?

### DISCUSSION THREAD

**Discussion by ethan-auroria** [WalletConnect/walletconnect-monorepo#4184] [Developer Support] [answered]
I want to disconnect the current wallet when user clicks on the logout button.  However, I want to invoked it from my code because there is a need to perform other clean up tasks at the same time, and I don't want users having to perform another action to disconnect the wallet.

There is a function `modal.disconnect()` as described in [this doc](https://docs.walletconnect.com/web3modal/javascript/actions), but when I call it, I get the error `disconnect is not a function` and the wallet is not disconnected.

**Comment by glitch-txs** [comment]
Are you using Wagmi?

**Reply by ethan-auroria** [reply]
I found out the solution, posted below. IMO the documentations need some serious reviewing. The same question was asked by someone else acouple of months ago and there wasn't any answer to it either before today.

**Comment by ethan-auroria** [comment]
Answering my own question because it is not present in the docs or anywhere else.

```javascript
	function onAccountChanged(account: GetAccountResult) {
		if (account.address !== undefined || account.isConnected) {
			account.connector?.disconnect();

			goto('/');
		}

		// I'm not sure what should be done here, because we are assuming that there is an account connected.
	}

	onMount(() => {
		const metadata = {...};
		const chains = [...];
		const wagmiConfig = defaultWagmiConfig({ chains, projectId: PUBLIC_WALLETCONNECT_PROJECT_ID, metadata });

		createWeb3Modal({ wagmiConfig, projectId: PUBLIC_WALLETCONNECT_PROJECT_ID, chains });
		unwatchAccount = watchAccount(onAccountChanged);

		return () => {
			// Remove callbacks.
		};
	});
```

**Reply by glitch-txs** [reply, marked as answer]
yep, you could also use this: https://1.x.wagmi.sh/core/actions/disconnect
I need to update docs since that method only belong to ethers implementation
